core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java

```diff
@@ -48,12 +51,23 @@ private LoggingDecorators() {}
     public static void logWhenComplete(
             Logger logger, RequestContext ctx,
             Consumer<RequestOnlyLog> requestLogger, Consumer<RequestLog> responseLogger) {
-        ctx.log().whenRequestComplete().thenAccept(requestLogger).exceptionally(e -> {
-            try (SafeCloseable ignored = ctx.push()) {
-                logger.warn("{} Unexpected exception while logging request: ", ctx, e);
-            }
-            return null;
-        });
+        final boolean isTransientService = isTransientService(ctx);
+        final CompletableFuture<RequestOnlyLog> requestCompletionFuture = ctx.log().whenRequestComplete();
+        if (!isTransientService) {
+            requestCompletionFuture.thenAccept(requestLogger).exceptionally(e -> {
+                try (SafeCloseable ignored = ctx.push()) {
+                    logger.warn("{} Unexpected exception while logging request: ", ctx, e);
+                }
+                return null;
+            });
+        } else {
+            requestCompletionFuture.exceptionally(e -> {
+                try (SafeCloseable ignored = ctx.push()) {
+                    logger.warn("{} Unexpected exception while logging request: ", ctx, e);
+                }
+                return null;
+            });
```

Comment:
This `.exceptionally()` call seems to be a duplicate. Could be simplified to:

```java
final CompletableFuture<RequestOnlyLog> requestCompletionFuture;
if (!isTransientService) {
    requestCompletionFuture = ctx.log().whenRequestComplete().thenAccept(...);
} else {
    requestCompletionFuture = ctx.log().whenRequestComplete();
}
requestCompletionFuture.exceptionally(...);
```

Reply:
I also considered. But, even if `whenRequestComplete()` returns `CompletableFuture<RequestOnlyLog>`, `whenRequestComplete().thenAccept(...)` returns `CompletableFuture<Void>`. Meanwhile, raw use of `CompletableFuture` is ambiguous and using a template seems complicated. 🤔

Reply:
Then, perhaps we can use `thenApply`? 😄
```java
requestCompletionFuture = ctx.log().whenRequestComplete().thenApply(log -> {
    requestLogger.accept(log);
    return log;
});
```
It's a bit verbose though. 😄